src/Sentry.Unity/SentryUnity.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using Sentry.Extensibility;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Sentry.Unity
+{
+    public sealed class SentryUnity : IDisposable
+    {
+        private static SentryUnity? Instance;
+
+        internal readonly ErrorTimeDebounce ErrorTimeDebounce = new(TimeSpan.FromSeconds(1));
+        internal readonly LogTimeDebounce LogTimeDebounce = new(TimeSpan.FromSeconds(1));
+        internal readonly WarningTimeDebounce WarningTimeDebounce = new(TimeSpan.FromSeconds(1));
+
+        internal UnitySentryOptions? Options { get; private set; }
+
+        internal IEventCapture EventCapture { get; set; } = new EventCapture();
+
+        public static SentryUnity Init(UnitySentryOptions unitySentryOptions, Action<SentryUnity>? unityConfigure = null)
+        {
+            // Force dispose if new Init
+            if (Instance != null)
+            {
+                // TODO: discuss
+                // using the logger of a new instance
+                unitySentryOptions.Logger?.Log(SentryLevel.Warning, "RE-INIT");
+
+                Instance.Dispose();
+            }
+
+            SentrySdk.Init(unitySentryOptions.ToSentryOptions());
+
+            Instance = new SentryUnity { Options = unitySentryOptions }
+                .WithApplicationEvents()
+                .FinalizeSetup(unitySentryOptions.Logger);
+            unityConfigure?.Invoke(Instance);
+            return Instance;
+        }
+
+        public static SentryUnity Init(
+            Action<UnitySentryOptions> unitySentryOptionsConfigure,
+            Action<SentryUnity>? unityConfigure = null)
+        {
+            var unitySentryOptions = new UnitySentryOptions();
+            unitySentryOptionsConfigure.Invoke(unitySentryOptions);
+            return Init(unitySentryOptions, unityConfigure);
+        }
+
+        private SentryUnity WithApplicationEvents()
+        {
+            Application.logMessageReceived += OnLogMessageReceived;
+            Application.quitting += OnQuitting;
+
+            return this;
+        }
+
+        private SentryUnity FinalizeSetup(IDiagnosticLogger? diagnosticLogger)
+        {
+            var data = SceneManager.GetActiveScene().name is { } name
+                ? new Dictionary<string, string> {{"scene", name}}
+                : null;
+
+            SentrySdk.AddBreadcrumb("BeforeSceneLoad", data: data);
+
+            diagnosticLogger?.Log(SentryLevel.Debug, "Complete Sentry SDK initialization.");
+
+            return this;
+        }
+
+        private void OnLogMessageReceived(string condition, string stackTrace, LogType type)

```

Comment:
This one I meant can live on its own class like: `UnityApplicationLoggingIntegration` and could implement `ISdkIntegration`.

We add it like:

```
Sentry.Init(o => 
  o.Dsn = "";
  o.AddIntegration(new UnityApplicationLoggingIntegration(o);
);
```